Sources/MessagingInApp/Views/GistInlineInAppMessageView.swift

```diff
@@ -15,20 +15,255 @@ public protocol GistInlineMessageUIViewDelegate: AnyObject {
+            refreshViewListener?()
+        }
+
+        guard let elementId = elementId else {
+            return // we cannot check if a message is available until element id set on View.
+        }
+
+        let queueOfMessagesForGivenElementId = state.inlineMessagesInQueue(forElementId: elementId)
+        let messageAvailableToDisplay = queueOfMessagesForGivenElementId.first { !hasBeenPreviouslyShown($0) }
+
+        if !forceShowNextMessage, isRenderingOrDisplayingAMessage {
+            // We are already displaying or rendering a messsage. Do not show another message until the current message is closed.
+            // The main reason for this is when a message is tracked as "opened", the Gist backend will not return this message on the next fetch call.
+            // We want to coninue showing a message even if the fetch no longer returns the message and the message is currently visible.
+            return
+        }
+
+        if let messageAvailableToDisplay {
+            displayInAppMessage(state: state, message: messageAvailableToDisplay)
+        } else {
+            stopShowingMessageAndCleanup()
+            delegate?.onNoMessageToDisplay()
+        }
+    }
+
+    // Function to check if a message has been previously shown
+    func hasBeenPreviouslyShown(_ message: Message) -> Bool {
+        previouslyShownMessages.contains { $0.queueId == message.queueId }
+    }
+
+    private func displayInAppMessage(state: InAppMessageState, message: Message) {
+        // If this function gets called a lot in a short amount of time (eventbus triggers multiple events), the display animation does not look as expected.
+        // To fix this, exit early if display has already been triggered.
+        if let currentlyShownMessage = state.currentMessageState.message, currentlyShownMessage.queueId == message.queueId {
+            return // already showing this message or in the process of showing it.
+        }
+
+        // If a different message is currently being shown, we want to replace the currently shown message with new message.
+        if isRenderingOrDisplayingAMessage {
+            delegate?.willChangeMessage(newTemplateId: message.messageId) {
+                // After the delegate is done, cleanup resources since we no longer need to show the previous message.
+                self.stopShowingMessageAndCleanup()
+                self.beginShowing(state: state, message: message)
+            }
+        } else {
+            beginShowing(state: state, message: message)
+        }
+    }
+
+    // Call when you want to begin the process of showing a new message.
+    private func beginShowing(state: InAppMessageState, message: Message) {
+        // Create a new manager for this new message to display and then display the manager's WebView.
+        let newInlineMessageManager = InlineMessageManager(state: state, message: message)
+        newInlineMessageManager.inlineMessageDelegate = self
```

Comment:
Removed setting [following delegate from here](https://github.com/customerio/customerio-ios/blob/111d9d915820fc19c96ecbc802b21823b5613f3e/Sources/MessagingInApp/Views/GistInlineInAppMessageView.swift#L176-L178) since we no longer have the `delegate` in `InlineMessageManager`. Not yet sure if any replacement is needed

```swift
// Gist class is what Modal messages use as the modal message manager delegate.
// So we can re-use modal message logic, set the Gist class for inline managers, too.
newInlineMessageManager.delegate = Gist.shared
```